### PR TITLE
Add branch protection commands

### PR DESCRIPTION
**Pull request by maximede**

### CODE REVIEW

File: src/base.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import {cli} from 'cli-ux';
+
+import {GithubClient} from './service/github-client';
+
+export default abstract class extends Command {
+  static flags = {
+    help: flags.help({char: 'h'}),
+    ghToken: flags.string({char: 't', description: 'Github Token for authentication'}),
+    owner: flags.string({char: 'o', required: true, default: 'uw-it-edm'}),
+    //dryRun: flags.boolean({char: 'd'})
```
Comment (agagne) [inline src/base.ts:11]: no dryRun?

File: src/commands/add-branch-protection.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    ...Command.args,
+  ];
+
+  async run() {
+    const {args, flags} = this.parse(AddBranchProtection);
+    cli.log(`Working with repo ${flags.owner}/${args.repo}`);
+
+    let repo = await super.githubClient.getRepo(flags.owner, args.repo);
+
+    //TODO create repo if doesn't exist ?  ( https://octokit.github.io/rest.js/#api-Repos-create. Turn on `auto_init` and created develop and master branch using https://octokit.github.io/rest.js/#api-Gitdata-createReference
```
Comment (agagne) [inline src/commands/add-branch-protection.ts]: I don't think so, I think you'd want to be explicit in creating a repo

File: src/commands/enforce-admin-protection.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+  async run() {
+    const {args, flags} = this.parse(EnforceAdminProtection);
+    cli.log(`Working with repo ${flags.owner}/${args.repo}`);
+
+    let repo = await super.githubClient.getRepo(flags.owner, args.repo);
+
+    cli.log(`found repo ${repo.name}`);
+
+    //TODO create repo if doesn't exist ?  ( https://octokit.github.io/rest.js/#api-Repos-create. Turn on `auto_init` and created develop and master branch using https://octokit.github.io/rest.js/#api-Gitdata-createReference
```
Comment (agagne) [inline src/commands/enforce-admin-protection.ts]: Wouldn't we want to be explicit about creating a repo?

File: README.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+<!-- usagestop -->
+# Commands
+<!-- commands -->
+* [`edm-repos add-branch-protection REPO`](#edm-repos-add-branch-protection-repo)
+* [`edm-repos enforce-admin-protection REPO`](#edm-repos-enforce-admin-protection-repo)
+* [`edm-repos help [COMMAND]`](#edm-repos-help-command)
+
+## `edm-repos add-branch-protection REPO`
+
+describe the command here
```
Comment (huiaic) [inline README.md]: remove please

File: README.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+  found repo uw-it-edm/test-repo -- public
+  Added develop branch protection. Response status 200
+  Added master branch protection. Response status 200
+```
+
+_See code: [src/commands/add-branch-protection.ts](https://github.com/uw-it-edm/edm-repos-cli/blob/v0.0.0/src/commands/add-branch-protection.ts)_
+
+## `edm-repos enforce-admin-protection REPO`
+
+describe the command here
```
Comment (huiaic) [inline README.md]: remove please

File: src/base.ts (hunk 1 above)
Comment (maximede) [reply src/base.ts:11]: might be more of a pain to implement that I originally thought

**Review by agagne — APPROVED** [review]

**Pull request merged**